includes/Settings.php

```diff
@@ -407,15 +410,73 @@ public function get_setting( string $key, $default = false ) {
 			 *
 			 * @var string|array<int|string,mixed>|bool
 			 */
-			return get_option( $key, $default );
+			$option = get_option( $key, $default );
+			if ( $option === $default ) {
+				return $option;
+			}
+		} else {
+			/**
+			 * Setting value.
+			 *
+			 * @var string|array<int|string,mixed>|bool
+			 */
+			$option = get_option( $key );
+		}
+
+		$settings = $this->get_registered_options();
+		if ( isset( $settings[ $key ] ) ) {
+			$value = rest_sanitize_value_from_schema( $option, $settings[ $key ] );
+			if ( is_wp_error( $value ) ) {
+				return $option;
+			}
+			/**
+			 * Setting value.
+			 *
+			 * @var string|array<int|string,mixed>|bool
+			 */
+			$option = $value;
+		}
+
+		return $option;
+	}
+
+	/**
+	 * Retrieves all of the registered options for the Settings API.
+	 *
+	 * @since 1.27.0
+	 *
+	 * @return array<string, array<string,string>> Array of registered options.
+	 */
+	protected function get_registered_options(): array {
```

Comment:
Similar to but not the same as - https://github.com/WordPress/wordpress-develop/blob/302ba4f45f96490f5947f324228f20e0cbe28583/src/wp-includes/rest-api/endpoints/class-wp-rest-settings-controller.php#L211-L267

Reply:
Why do we need a custom version?

What are the exact differences?

Would be good to at least reference the original & the differences in the docblock.

Reply:
Added a comment and link. It is not a little different. It is mostly because that method is part of settings rest api, so not easy to reuse the original method.